src/models/parsers/ParsingContext.ts

```diff
@@ -0,0 +1,76 @@
+import { Message } from "discord.js";
+import { InvalidTypeParseError, NotEnoughArgParseError } from "./errors";
+
+export class ParsingContext {
+    public readonly message: Message;
+    private readonly args: readonly string[];
+    private readonly from: number;
+    private readonly to: number;
+    private cur: number;
+
+    public constructor(message: Message, args: readonly string[], from = 0, to?: number) {
+        this.message = message;
+        this.args = args;
+        this.from = Math.max(from, 0);
+        this.to = Math.min(Math.max(to ?? args.length, from), args.length);
```

Comment:
```suggestion
        this.to = to ? clamp(to, from, args.length) : args.length;
```
Create a utils function `clamp`